File: examples/golang/device_management.go
package main

import (
	"fmt"
	"strings"
)

/**
 * Contoh Kode: Device Management (Session Management)
 * 
 * File ini berisi contoh implementasi untuk mengelola device/session WhatsApp
 * termasuk membuat device baru, mendapatkan QR code, dan mengecek status device.
 */

func main() {
	fmt.Println("=== WACloud - Device Management Examples ===\n\n")

	// ============================================
	// 1. Membuat Device Baru
	// ============================================
	fmt.Println("1. Membuat Device Baru")
	fmt.Println(strings.Repeat("-", 50))

	deviceData := map[string]interface{}{
		"name":         "Device Utama",
		"phone_number": "81234567890", // Format: tanpa leading 0, 9-13 digit
	}

	response, err := MakeRequest("POST", "/devices", deviceData)
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		return
	}

	DisplayResponse(response)

	if response.Success {
		if deviceID, ok := response.Data["id"].(string); ok {
			fmt.Printf("Device ID: %s\n", deviceID)
			if status, ok := response.Data["status"].(string); ok {
				fmt.Printf("Status: %s\n\n", status)
			}

			// ============================================
			// 2. Mendapatkan QR Code untuk Pairing
			// ============================================
			fmt.Println("2. Mendapatkan QR Code untuk Pairing")
			fmt.Println(strings.Repeat("-", 50))

			qrResponse, err := MakeRequest("GET", "/devices/"+deviceID+"/pair", nil)
			if err != nil {
				fmt.Printf("Error: %v\n", err)
				return
			}

			DisplayResponse(qrResponse)

			if qrResponse.Success {
				if qrCode, ok := qrResponse.Data["qr_code"].(string); ok {
					if len(qrCode) > 50 {
						fmt.Printf("QR Code (Base64): %s...\n", qrCode[:50])
					} else {
						fmt.Printf("QR Code (Base64): %s\n", qrCode)
					}
					if expiresAt, ok := qrResponse.Data["expires_at"].(string); ok {
						fmt.Printf("Expires At: %s\n", expiresAt)
					}
					fmt.Println("\nUntuk menampilkan QR code di browser:")
					fmt.Printf("<img src='%s' alt='QR Code' />\n\n", qrCode)
				}
			}

			// ============================================
			// 3. Mendapatkan Status Device
			// ============================================
			fmt.Println("3. Mendapatkan Status Device")
			fmt.Println(strings.Repeat("-", 50))

			statusResponse, err := MakeRequest("GET", "/devices/"+deviceID+"/status", nil)
			if err != nil {
				fmt.Printf("Error: %v\n", err)
				return
			}

			DisplayResponse(statusResponse)

			// ============================================
			// 4. Mendapatkan Detail Device
			// ============================================
			fmt.Println("4. Mendapatkan Detail Device")
			fmt.Println(strings.Repeat("-", 50))

			detailResponse, err := MakeRequest("GET", "/devices/"+deviceID, nil)
			if err != nil {
				fmt.Printf("Error: %v\n", err)
				return
			}

			DisplayResponse(detailResponse)
		}
	} else {
		fmt.Println("Gagal membuat device. Pastikan API_KEY sudah benar.\n")
	}

	// ============================================
	// 5. Mendapatkan Daftar Semua Device
	// ============================================
	fmt.Println("5. Mendapatkan Daftar Semua Device")
	fmt.Println(strings.Repeat("-", 50))

	devicesResponse, err := MakeRequest("GET", "/devices", nil)
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		return
	}

	DisplayResponse(devicesResponse)

	fmt.Println("\n=== Selesai ===")
}

File: examples/php/device-management.php
<?php
/**
 * Contoh Kode: Device Management (Session Management)
 * 
 * File ini berisi contoh implementasi untuk mengelola device/session WhatsApp
 * termasuk membuat device baru, mendapatkan QR code, dan mengecek status device.
 */

require_once 'config.php';

echo "=== WACloud - Device Management Examples ===\n\n";

// ============================================
// 1. Membuat Device Baru
// ============================================
echo "1. Membuat Device Baru\n";
echo str_repeat('-', 50) . "\n";

$deviceData = [
    'name' => 'Device Utama',
    'phone_number' => '81234567890' // Format: tanpa leading 0, 9-13 digit
];

$response = makeRequest('POST', '/devices', $deviceData);
displayResponse($response);

if ($response['success'] && isset($response['data']['id'])) {
    $deviceId = $response['data']['id'];
    echo "Device ID: {$deviceId}\n";
    echo "Status: {$response['data']['status']}\n\n";
    
    // ============================================
    // 2. Mendapatkan QR Code untuk Pairing
    // ============================================
    echo "2. Mendapatkan QR Code untuk Pairing\n";
    echo str_repeat('-', 50) . "\n";
    
    $qrResponse = makeRequest('GET', "/devices/{$deviceId}/pair");
    displayResponse($qrResponse);
    
    if ($qrResponse['success'] && isset($qrResponse['data']['qr_code'])) {
        $qrCode = $qrResponse['data']['qr_code'];
        $expiresAt = $qrResponse['data']['expires_at'] ?? 'N/A';
        
        echo "QR Code (Base64): " . substr($qrCode, 0, 50) . "...\n";
        echo "Expires At: {$expiresAt}\n";
        echo "\nUntuk menampilkan QR code di browser:\n";
        echo "<img src='{$qrCode}' alt='QR Code' />\n\n";
    }
    
    // ============================================
    // 3. Mendapatkan Status Device
    // ============================================
    echo "3. Mendapatkan Status Device\n";
    echo str_repeat('-', 50) . "\n";
    
    $statusResponse = makeRequest('GET', "/devices/{$deviceId}/status");
    displayResponse($statusResponse);
    
    // ============================================
    // 4. Mendapatkan Detail Device
    // ============================================
    echo "4. Mendapatkan Detail Device\n";
    echo str_repeat('-', 50) . "\n";
    
    $detailResponse = makeRequest('GET', "/devices/{$deviceId}");
    displayResponse($detailResponse);
} else {
    echo "Gagal membuat device. Pastikan API_KEY sudah benar.\n";
}

// ============================================
// 5. Mendapatkan Daftar Semua Device
// ============================================
echo "5. Mendapatkan Daftar Semua Device\n";
echo str_repeat('-', 50) . "\n";

$devicesResponse = makeRequest('GET', '/devices');
displayResponse($devicesResponse);

if ($devicesResponse['success'] && isset($devicesResponse['data'])) {
    echo "Total Devices: " . count($devicesResponse['data']) . "\n";
    foreach ($devicesResponse['data'] as $device) {
        echo "- {$device['name']} (ID: {$device['id']}, Status: {$device['status']})\n";
    }
}

echo "\n=== Selesai ===\n";

